core/src/storage/impls/merkle_patricia_trie/mpt_cursor.rs

```diff
@@ -726,19 +741,32 @@ pub trait PathNodeTrait<Mpt: GetReadMpt>:
         cursor: &mut Cursor,
     ) -> Result<Self> {
         let mut mpt = cursor.take_mpt();
-        let root_trie_node = cursor.load_node_wrapper(
+        let root_trie_node = match cursor.load_node_wrapper(
             mpt.as_mut_assumed_owner(),
             &CompressedPathRaw::default(),
-        )?;
-
-        let supposed_merkle_root = mpt.as_ref_assumed_owner().get_merkle_root();
-        assert_eq!(
-            root_trie_node.get_merkle(),
-            supposed_merkle_root,
-            "loaded root trie node merkle hash {:?} != supposed merkle hash {:?}",
-            root_trie_node.get_merkle(),
-            supposed_merkle_root,
-        );
+        ) {
+            Ok(root_trie_node) => {
+                let supposed_merkle_root =
+                    mpt.as_ref_assumed_owner().get_merkle_root();
+                assert_eq!(
+                    root_trie_node.get_merkle(),
+                    supposed_merkle_root,
+                    "loaded root trie node merkle hash {:?} != supposed merkle hash {:?}",
+                    root_trie_node.get_merkle(),
+                    supposed_merkle_root,
+                );
+                root_trie_node
+            }
+            Err(Error(ErrorKind::SnapshotMPTTrieNodeNotFound, _)) => {
```

Comment:
The best place is to create a special case for SnapshotMpt, when the maybe_db is None in SnapshotDbSqlite. Please add a FIXME then I'll fix it in the next PR